app/Http/Controllers/ProductsController.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Service\ProductService;
+use Carbon\Carbon;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Response;
+use Illuminate\Support\Collection;
+
+class ProductsController extends Controller
+{
+    private ProductService $productService;
+
+    public function __construct(ProductService $productService)
+    {
+        $this->productService = $productService;
+    }
+
+    public function getProducts(): JsonResponse
+    {
+        $data = $this->productService->parseProducts();
+        /** @var Collection $products */
+        $products = $data['products'];
```

Comment:
Maybe it would be better to compute the relevant data in the ProductService, so that you could easily call it from another Controller if needed